contracts/token/ERC721/extensions/draft-ERC721Royalty.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: MIT
+// OpenZeppelin Contracts v4.4.0 (token/ERC721/extensions/draft-ERC721Royalty.sol)
+
+pragma solidity ^0.8.0;
+
+import "../ERC721.sol";
+import "./draft-IERC721Royalty.sol";
+import "../../../utils/introspection/ERC165.sol";
+
+/**
+ * @dev Implementation of the ERC721 Royalty extension allowing royalty information to be stored and retrieved, as defined in
+ * https://eips.ethereum.org/EIPS/eip-2981[EIP-2981].
+ *
+ * Adds the {_setTokenRoyalty} methods to set the token royalty information, and {_setGlobalRoyalty} method to set a global
+ * royalty information.
+ *
+ * _Available since v4.5._
+ */
+abstract contract ERC721Royalty is IERC721Royalty, ERC721 {
+    struct RoyaltyInfo {
+        address receiver;
+        uint256 royaltyFraction;
+    }
+
+    RoyaltyInfo private _globalRoyaltyInfo;
+    mapping(uint256 => RoyaltyInfo) private _tokenRoyaltyInfo;
+
+    /**
+     * @dev Sets tokens royalties
+     *
+     * Requirements:
+     * - `tokenId` must be already mined.
+     * - `receiver` cannot be the zero address.
+     * - `fraction` must indicate the percentage fraction using two decimals.
+     */
+    function _setTokenRoyalty(
+        uint256 tokenId,
+        address receiver,
+        uint256 fraction
+    ) internal virtual {
+        require(fraction > 0, "ERC2981: Royalty percentage is too low");
```

Comment:
I think this function could be used to set but also to update ... and updating in a way that sets the fee to 0 might be acceptable.

Also, if the global fee is not 0, maybe setting a custom token fee to 0 makes sens.

So I would remove that check.

Reply:
Yes, I'll remove it, and also they can use to update the royalty not just set it.